client/me/account/toggle-sites-as-landing-page.tsx

```diff
@@ -0,0 +1,62 @@
+import { localizeUrl } from '@automattic/i18n-utils';
+import { ToggleControl as OriginalToggleControl } from '@wordpress/components';
+import { createInterpolateElement } from '@wordpress/element';
+import { useTranslate } from 'i18n-calypso';
+import { useDispatch, useSelector } from 'react-redux';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { successNotice } from 'calypso/state/notices/actions';
+import { savePreference } from 'calypso/state/preferences/actions';
+import { getPreference, isSavingPreference } from 'calypso/state/preferences/selectors';
+
+const ToggleControl = OriginalToggleControl as React.ComponentType<
+	OriginalToggleControl.Props & {
+		disabled?: boolean;
+	}
+>;
+
+function ToggleSitesAsLandingPage() {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const useSitesAsLandingPage = useSelector(
+		( state ) => getPreference( state, 'sites-landing-page' )?.useSitesAsLandingPage
+	);
+	const isSaving = useSelector( ( state ) => isSavingPreference( state ) );
+
+	async function handleToggle( isChecked: boolean ) {
+		const preference = { useSitesAsLandingPage: isChecked, updatedAt: Date.now() };
+		await dispatch( savePreference( 'sites-landing-page', preference ) );
+
+		dispatch(
+			successNotice( translate( 'Settings saved successfully!' ), {
+				id: 'sites-landing-page-save',
+				duration: 10000,
+			} )
+		);
+
+		dispatch(
+			recordTracksEvent( 'calypso_sites_as_landing_page_toggled', {
```

Comment:
I'm thinking maybe we should change the event names for consistency: https://github.com/Automattic/wp-calypso/pull/70512

So maybe this should be

```suggestion
			recordTracksEvent( 'calypso_sites_dashboard_landing_page_setting_toggle', {
```

Or maybe `_click`?

It might depend on the answer to this p1669699187812099-slack-C0347E545HR

Reply:
I'd say `calypso_settings_sites_dashboard_landing_page_toggle`. It's good to understand the origin of the event -- it's almost always the prefix.